crates/aptos-api-tester/src/main.rs

```diff
@@ -0,0 +1,250 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use std::future::Future;
+
+use anyhow::{anyhow, Result};
+use aptos_api_types::U64;
+use aptos_rest_client::{Account, Client, FaucetClient};
+use aptos_sdk::coin_client::CoinClient;
+use aptos_sdk::token_client::TokenClient;
+use aptos_sdk::types::LocalAccount;
+use aptos_types::account_address::AccountAddress;
+use once_cell::sync::Lazy;
+use url::Url;
+
+// network urls
+// static DEVNET_NODE_URL: Lazy<Url> =
+//     Lazy::new(|| Url::parse("https://fullnode.devnet.aptoslabs.com").unwrap());
+// static DEVNET_FAUCET_URL: Lazy<Url> =
+//     Lazy::new(|| Url::parse("https://faucet.devnet.aptoslabs.com").unwrap());
+static TESTNET_NODE_URL: Lazy<Url> =
+    Lazy::new(|| Url::parse("https://fullnode.testnet.aptoslabs.com").unwrap());
+static TESTNET_FAUCET_URL: Lazy<Url> =
+    Lazy::new(|| Url::parse("https://faucet.testnet.aptoslabs.com").unwrap());
+
+// static accounts to use
+static TEST_ACCOUNT_1: Lazy<AccountAddress> = Lazy::new(|| {
+    AccountAddress::from_hex_literal(
+        "0xf6cee5c359839a321a08af6b8cfe4a6e439db46f4942e21b6845aa57d770efdb",
+    )
+    .unwrap()
+});
+
+#[derive(Debug)]
+enum TestResult {
+    Success,
+    Fail(&'static str),
+    Error(anyhow::Error),
+}
+
+async fn handle_result<Fut: Future<Output = TestResult>>(fut: Fut) -> TestResult {
+    let result = fut.await;
+    println!("{:?}", result);
+
+    result
+}
+
+/// Tests new account creation. Checks that:
+///   - account data exists
+///   - account balance reflects funded amount
+async fn test_newaccount(
+    client: &Client,
+    account: &LocalAccount,
+    amount_funded: u64,
```

Comment:
Not sure what context you're going to call this from- but I believe the faucets only ever send back a max amount of APT and it's not considered an error. Like if you request 10 it will still give you 1.

You might want to find the hard-coded max fixed amount and assert that the post-fund balance is the max fixed amount rather than the requested amount in the case that the requested amount is greater than the max amount that can be sent from the faucet

Reply:
yep confirmed that 1APT is max, thanks!